src/commands/tui/ls.rs

```diff
@@ -217,18 +217,20 @@ impl<'a, P: ProgressBars, S: IndexedFull> Snapshot<'a, P, S> {
                                     if let Ok(data) = OpenFile::from_node(self.repo, node).read_at(
                                         self.repo,
                                         0,
-                                        node.meta.size.min(1000000).try_into().unwrap(),
+                                        node.meta.size.min(1_000_000).try_into().unwrap(),
                                     ) {
-                                        let content = String::from_utf8_lossy(&data);
-                                        let lines = content.lines().count();
-                                        let path = self.path.join(node.name());
-                                        let path = path.display();
-                                        self.current_screen =
-                                            CurrentScreen::ShowFile(popup_scrollable_text(
-                                                format!("{}:/{path}", self.snapshot.id),
-                                                &content,
-                                                (lines + 1).min(40).try_into().unwrap(),
-                                            ));
+                                        // viewing is only supported for text files
+                                        if let Ok(content) = String::from_utf8(data.to_vec()) {
```

Comment:
What do you think about a popup that warns that the `view` action failed because the file isn't full utf8?

Reply:
Agree - but IMO this can go with other future user improvements. It may be a bit confusing but is doing no harm...